internal/api/external.go

```diff
@@ -187,7 +187,20 @@ func (a *API) internalExternalProviderCallback(w http.ResponseWriter, r *http.Re
 	if err != nil {
 		return err
 	}
+
 	userData := data.userData
+	if len(userData.Emails) <= 0 {
+		return internalServerError("Error getting user email from external provider")
+	}
```

Comment:
[this will be done in a FLUP PR since the current one is quite big] 

now that the email check is done at the top level, we can use the `ALLOW_UNVERIFIED_EMAIL_SIGN_INS` config to bypass returning this error